src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -0,0 +1,284 @@
+import { classMap } from "lit-html/directives/classMap.js";
+import roundSliderStyle from "round-slider/dist/roundslider.min.css";
+
+import "../../../components/ha-card.js";
+import "../../../components/ha-icon.js";
+import "../../../resources/jquery.roundslider";
+
+import { HomeAssistant } from "../../../types.js";
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+const thermostatConfig = {
+  radius: 150,
+  min: 50,
+  max: 90,
+  step: 1,
+  circleShape: "pie",
+  sliderType: "min-range",
+  value: 70,
+  startAngle: 315,
+  width: 5,
+  lineCap: "round",
+  handleSize: "+10",
+  showTooltip: true,
+};
+
+const modeIcons = {
+  auto: "mdi:autorenew",
+  heat: "hass:fire",
+  cool: "hass:snowflake",
+  off: "mdi:fan-off",
+};
+
+interface Config extends LovelaceConfig {
+  entity: string;
+}
+
+let loaded: Promise<void>;
+
+export class HuiThermostatCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected hass?: HomeAssistant;
+  protected config?: Config;
+  protected roundSlider: any;
+
+  static get properties() {
+    return {
+      hass: {},
+      config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 4;
+  }
+
+  public setConfig(config: Config) {
+    if (!config.entity || config.entity.split(".")[0] !== "climate") {
+      throw new Error("Specify an entity from within the climate domain.");
+    }
+
+    this.config = config;
+  }
+
+  protected render() {
+    const stateObj = this.hass!.states[this.config!.entity];
+    const mode = stateObj.attributes.operation_mode;
+    return html`
+      ${this.renderStyle()}
+      <ha-card
+        class="${mode}">
+        <div
+          .hass="${this.hass}"
+          .config="${this.config}"
+          .localize="${this.localize}"
+          id="thermostat"
+        ></div>
+      </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps) {
+    if (changedProps.get("hass")) {
+      return changedProps.get("hass").states[this.config!.entity] !==
+        this.hass!.states[this.config!.entity]
+        ? true
+        : false;
+    }
+    return changedProps;
+  }
+
+  protected firstUpdated(changedProps) {
+    const stateObj = this.hass!.states[this.config!.entity];
+
+    $("#thermostat", this.shadowRoot).roundSlider({
+      radius: this.clientWidth / 3,
+      min: stateObj.attributes.min_temp,
+      max: stateObj.attributes.max_temp,
+      step: thermostatConfig.step,
+      circleShape: thermostatConfig.circleShape,
+      sliderType: thermostatConfig.sliderType,
+      startAngle: thermostatConfig.startAngle,
+      width: thermostatConfig.width,
+      handleSize: thermostatConfig.handleSize,
+      lineCap: thermostatConfig.lineCap,
+      showTooltip: thermostatConfig.showTooltip,
+      editableTooltip: false,
+      tooltipFormat: this.tooltip.bind(this),
+      change: this._setTemperature.bind(this),
```

Comment:
Prefer arrow over bind: `temp => this._setTemperature(temp)`

Reply:
Yea. After our conversation yesterday about arrow I thought about those calls